qiskit/pulse/utils.py

```diff
@@ -0,0 +1,119 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Pulse utilities.
+"""
+import warnings
+
+from typing import List
+
+import numpy as np
+
+from .channels import AcquireChannel, MemorySlot
+from .commands import Acquire, AcquireInstruction
+from .interfaces import ScheduleComponent
+from .schedule import Schedule
+from .cmd_def import CmdDef
+
+
+def align_measures(schedules: List[ScheduleComponent], cmd_def: CmdDef) -> Schedule:
+    """Return new schedules where measurements occur at the same physical time. Minimum measurement
+    wait time (to allow for calibration pulses) is enforced.
+    This is only defined for schedules that are acquire-less or acquire-final per channel: a
+    schedule with pulses or acquires occuring on a channel which has already had an acquire will
+    throw an error.
+
+    Args:
+        schedules: Collection of schedules to be aligned together
+        cmd_def: Command definition list
+    Returns:
+        Schedule
+    Raises:
+        ValueError: if an acquire or pulse is encountered on a channel that has already been part
+                    of an acquire
+    """
+    # Need time to allow for calibration pulses to be played for result classification
+    max_calibration_duration = 0
+    for qubits in cmd_def.cmd_qubits('u3'):
+        cmd = cmd_def.get('u3', qubits, np.pi, 0, np.pi)
```

Comment:
I think it might be nice to make this gate (`u3`) configurable, in case other basis gates are used. I suppose this could be done in one of two ways (or both). 
1. Allow the `cmd_def` gate to be used for calibration to be supplied as an argument in string form
2. Allow the parameter max_calibration_duration to be supplied as an argument. In this case the cmd_def would be ignored?